src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -73,27 +76,27 @@ public synchronized void write(byte[] data) throws IOException {
     }
 
     public void openFile() throws IOException {
-        GZIPFileDescriptor fileDescriptor = new GZIPFileDescriptor();
+        FileDescriptor fileDescriptor = new FileDescriptor();
 
         File folder = new File(basePath);
         if (!folder.exists() && !folder.mkdirs()) {
             throw new IOException(String.format("Failed to create new directory %s", folder.getPath()));
         }
 
-        String filePath = getFilePath.get() + ".gz";
+        String filePath = getFilePath.get() + (compressionType == CompressionType.zip ? ".zip" : ".gz");
```

Comment:
we can make it clearer

Maybe?
```
String fileExt = '';

switch (compressionType) {
    case (zip):
            fileExt = ".zip";
            brea;
     case (gzip):
            fileExt = ",gz";
      default:
            break;
}
```

Reply:
this is not the same logic..
i will make it a constructor parameter